src/adr/0011-remove-elasticsearch-document-source.md

```diff
@@ -27,6 +27,11 @@ document.
 Without `_source` field, flags update is two time slower, 99 percentile 4 time slower, and this impact negatively other 
 requests.
 
+Note please that `_source` allows admin flexibility like performing index level changes without downtime, amongst others:
+ - Increase shards
+ - Modifying replication factor
```

Comment:
are you sure these two things are not doable without `_source`? It sounds strange

Reply:
I'm sure that ES built in reindexing do not work without `_source`.

To quote https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-source-field.html :

```
Think before disabling the _source field

Users often disable the _source field without thinking about the consequences, and then live to regret it. If the _source field isn’t available then a number of features are not supported:

-   The update, update_by_query, and reindex APIs.
-   On the fly highlighting.
-   The ability to reindex from one Elasticsearch index to another, either to change mappings or analysis, or to upgrade an index to a new major version.
-   The ability to debug queries or aggregations by viewing the original document used at index time.
 -   Potentially in the future, the ability to repair index corruption automatically.
```

BTW beloved ElasticSearch evangelists quote in this discussion reindex as the only way to change shard&replica count.

Reply:
> BTW beloved ElasticSearch evangelist quote in this discussion reindex as the only way to change shard&replica count.

The link between reindex and shard+replica was not obvious but OK.

Reply:
Not my fault.